.github/workflows/collector-slim.yml

```diff
@@ -20,44 +20,69 @@ env:
   RHACS_ENG_IMAGE: quay.io/rhacs-eng/collector:${{ inputs.collector-tag }}
 
 jobs:
+  set-environment:
+    name: Set environment variables
+    runs-on: ubuntu-latest
+    outputs:
+      build-builder-image: ${{ steps.set-env.outputs.build-builder-image }}
+      collector-builder-tag: ${{ steps.set-env.outputs.collector-builder-tag }}
+
+    steps:
+      - name: Checks PR, main and release branches
+        id: set-env
+        run: |
+
+          if [[ "${{ github.event_name }}" != 'pull_request' ]]; then
+            echo 'build-builder-image=true' >> "$GITHUB_OUTPUT"
+            echo "collector-builder-tag=${DEFAULT_BUILDER_TAG}" >> "$GITHUB_OUTPUT"
+
+          elif [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+            #We have 2 options:
+            #- We build the builder from scratch and give it a custom tag
+            #- We use the existing cache
+            COLLECTOR_BUILDER_TAG="${DEFAULT_BUILDER_TAG}"
+            if [[ "${{ contains(github.event.pull_request.labels.*.name, 'build-builder-image') }}" == "true" ]]; then
+              COLLECTOR_BUILDER_TAG="${{ inputs.collector-tag }}"
+              echo 'build-builder-image=true' >> "$GITHUB_OUTPUT"
+            fi
+
+            echo "collector-builder-tag=$COLLECTOR_BUILDER_TAG" >> "$GITHUB_OUTPUT"
+
+            echo "COLLECTOR_APPEND_CID=true" >> "$GITHUB_ENV"
+
+            if [[ "${{ contains(github.event.pull_request.labels.*.name, 'address-sanitizer') }}" == "true" ]]; then
+              echo "ADDRESS_SANITIZER=true" >> "$GITHUB_ENV"
+            fi
+          fi
+
   build-collector-image:
     name: Build the collector slim image
+    needs: set-environment
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64, ppc64le]
+
+    env:
+      BUILD_BUILDER_IMAGE: ${{ needs.set-environment.outputs.build-builder-image }}
+      COLLECTOR_BUILDER_TAG: ${{ needs.set-environment.outputs.collector-builder-tag }}
+      ARCH: ${{ matrix.arch }}
 
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: true
 
-      - name: Checks for main and release branches
-        if: github.event_name != 'pull_request'
+      - name: Log env vars
         run: |
-          echo 'BUILD_BUILDER_IMAGE=true' >> "$GITHUB_ENV"
-          echo "COLLECTOR_BUILDER_TAG=${DEFAULT_BUILDER_TAG}" >> "$GITHUB_ENV"
+          echo "[LOG] - $BUILD_BUILDER_IMAGE"
+          echo "[LOG] - $COLLECTOR_BUILDER_TAG"
```

Comment:
[nitpick] Environment variables are already logged in GHA if you expand the command: 
![image](https://user-images.githubusercontent.com/20206636/232738457-512d6681-a2f1-48e9-8b48-61083577b61f.png)

Feel free to keep this step, but I would remove it.